components/uniset/src/builder.rs

```diff
@@ -0,0 +1,755 @@
+    }
+
+    #[test]
+    fn test_add_to_empty() {
+        let mut builder = UnicodeSetBuilder::new();
+        builder.add(0, 10);
+        assert_eq!(builder.intervals, vec![0, 10]);
+    }
+
+    #[test]
+    fn test_add_invalid() {
+        let mut builder = UnicodeSetBuilder::new();
+        builder.add(0, 0);
+        builder.add(5, 0);
+        assert_eq!(builder.intervals, vec![]);
+    }
+
+    #[test]
+    fn test_add_to_start() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(0, 5);
+        let expected = vec![0, 5, 10, 20, 40, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_start_overlap() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(0, 15);
+        let expected = vec![0, 20, 40, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_end() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(60, 70);
+        let expected = vec![10, 20, 40, 50, 60, 70];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_end_overlap() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(45, 70);
+        let expected = vec![10, 20, 40, 70];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_middle_no_overlap() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(25, 27);
+        let expected = vec![10, 20, 25, 27, 40, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_middle_inside() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(10, 20);
+        let expected = vec![10, 20, 40, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_middle_left_overlap() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(15, 25);
+        let expected = vec![10, 25, 40, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_middle_right_overlap() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(30, 40);
+        let expected = vec![10, 20, 30, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_full_encompass() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(0, 60);
+        let expected = vec![0, 60];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_to_partial_encompass() {
+        let mut builder = generate_tester(vec![10, 20, 40, 50]);
+        builder.add(0, 35);
+        let expected = vec![0, 35, 40, 50];
+        assert_eq!(builder.intervals, expected);
+    }
+
+    #[test]
+    fn test_add_unicodeset() {
```

Comment:
Can you make this test a wee bit more interesting? :)

Reply:
agreed and changed (originally was doing nothing more than testing the `generate_tester` function basically), changed to multi element set union test w/ overlaps and non overlaps (`[10, 20, 40, 50] + [5, 10, 22, 33, 44, 51]`)